anyschema/_anyschema.py

```diff
@@ -29,12 +29,120 @@
     import pandas as pd
     import polars as pl
     import pyarrow as pa
+    from narwhals.dtypes import DType
     from narwhals.typing import DTypeBackend
     from typing_extensions import Self
 
     from anyschema.typing import Adapter, IntoParserPipeline, Spec
 
 
+__all__ = ("AnySchema", "Field")
+
+
+class Field:
+    """A structured field descriptor.
+
+    Arguments:
```

Comment:
Is there a reason why `description` isn't an argument as well? It's common enough to be included IMO. Users inspecting `Field` would benefit from seeing it on a clean attribute. However a `@property` derived from metadata may work as well.

Reply:
For SQLAlchemy you wanna use `doc`; the other one is literally injected into SQL as a comment :) 

Taking advantage of docstrings is a great idea.

Reply:
Addressed in https://github.com/FBruzzesi/anyschema/pull/76/commits/4098bc51fc2eea596dec72bcd29e311e0e3d246a